tests/devices/unit_tests/test_focusing_mirror.py

```diff
@@ -0,0 +1,119 @@
+from threading import Timer
+from unittest.mock import DEFAULT, MagicMock, patch
+
+import pytest
+from ophyd.sim import NullStatus
+from ophyd.status import Status, StatusBase
+
+from dodal.devices.focusing_mirror import (
+    DEMAND_ACCEPTED_OK,
+    MirrorVoltageDevice,
+    VFMMirrorVoltages,
+)
+
+
+@pytest.fixture
+def vfm_mirror_voltages_with_set(vfm_mirror_voltages) -> VFMMirrorVoltages:
+    def not_ok_then_ok(_):
+        vfm_mirror_voltages._channel14_voltage_device._demand_accepted.sim_put(0)
+        Timer(
+            0.1,
+            lambda: vfm_mirror_voltages._channel14_voltage_device._demand_accepted.sim_put(
+                1
+            ),
+        ).start()
+        return DEFAULT
+
+    vfm_mirror_voltages._channel14_voltage_device._setpoint_v.set = MagicMock(
+        side_effect=not_ok_then_ok
+    )
+    return vfm_mirror_voltages
+
+
+@pytest.fixture
+def vfm_mirror_voltages_with_set_timing_out(vfm_mirror_voltages) -> VFMMirrorVoltages:
+    def not_ok(_):
+        vfm_mirror_voltages._channel14_voltage_device._demand_accepted.sim_put(0)
+        return DEFAULT
+
+    vfm_mirror_voltages._channel14_voltage_device._setpoint_v.set = MagicMock(
+        side_effect=not_ok
+    )
+    return vfm_mirror_voltages
+
+
+def test_mirror_set_voltage_sets_and_waits_happy_path(
+    vfm_mirror_voltages_with_set: VFMMirrorVoltages,
+):
+    vfm_mirror_voltages_with_set._channel14_voltage_device._setpoint_v.set.return_value = (
+        NullStatus()
+    )
+    vfm_mirror_voltages_with_set._channel14_voltage_device._demand_accepted.sim_put(
+        DEMAND_ACCEPTED_OK
+    )
+
+    status: StatusBase = vfm_mirror_voltages_with_set.voltage_channels[0].set(100)
+    status.wait()
+    vfm_mirror_voltages_with_set._channel14_voltage_device._setpoint_v.set.assert_called_with(
+        100
+    )
+    assert status.success
+
+
+def test_mirror_set_voltage_sets_and_waits_set_fail(
+    vfm_mirror_voltages_with_set: VFMMirrorVoltages,
+):
+    vfm_mirror_voltages_with_set._channel14_voltage_device._setpoint_v.set.return_value = Status(
+        success=False, done=True
+    )
+
+    status: StatusBase = vfm_mirror_voltages_with_set.voltage_channels[0].set(100)
+    try:
+        status.wait()
+    except Exception:
+        pass
+
+    assert not status.success
+
+
+@patch("dodal.devices.focusing_mirror.DEFAULT_SETTLE_TIME_S", 3)
+def test_mirror_set_voltage_sets_and_waits_settle_timeout_expires(
+    vfm_mirror_voltages_with_set_timing_out: VFMMirrorVoltages,
+):
+    vfm_mirror_voltages_with_set_timing_out._channel14_voltage_device._setpoint_v.set.return_value = (
+        NullStatus()
+    )
+
+    status: StatusBase = vfm_mirror_voltages_with_set_timing_out.voltage_channels[
+        0
+    ].set(100)
+
+    actual_exception = None
+    try:
+        status.wait()
+    except Exception as e:
+        actual_exception = e
```

Comment:
Should: I think this removes the need for the assert too
```suggestion
    with pytest.raises(Exception) as e:
        status.wait()
```